levelmap.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 Siôn le Roux.  All rights reserved.
+// Use of this source code is subject to an MIT-style
+// licence which can be found in the LICENSE file.
+
+package main
+
+import (
+	"image"
+	"math"
+
+	"github.com/fzipp/astar"
+)
+
+type LevelMap [][]int
+
+
+func CreateMap(w, h int) LevelMap {
+	lmap := make([][]int, h)
+	for i := range lmap {
+		lmap[i] = make([]int, w)
+	}
+	
+	return lmap
+}
+
+// Neighbours implements the astar.Graph interface
+func (m LevelMap) Neighbours(p image.Point) []image.Point {
+	offsets := []image.Point{
+		image.Pt(0, -1), // North
+		image.Pt(1, 0),  // East
+		image.Pt(0, 1),  // South
+		image.Pt(-1, 0), // West
+	}
+	offsetsDiag := []image.Point{
+		image.Pt(1, -1),  // NorthEast
+		image.Pt(1, 1),   // SouthEast
+		image.Pt(-1, 1),  // SouthWest
+		image.Pt(-1, -1), // NorthWest
+	}
+	res := make([]image.Point, 0, 8)
```

Comment:
These things look actually like `consts` that exist only to be referenced later (e.g. by name) but are not expected to vary later, am I right?

Reply:
Yes. What to do with it then?

Reply:
Nothing I guess 😂 you can't make array constants, life is life.

OTOH you could give them all names if you like:

```go
const N = something
const NE = idk
cost E = ...etc
```

no idea...